src/routers/secure/scrape.py

```
@@ -185,6 +186,63 @@ def initialize_downloader(downloader: Downloader):
+    if not container or not container.cached:
+        raise HTTPException(status_code=400, detail=f"Torrent is not cached, please try another {source_description}")
+
+    session = session_manager.create_session(item_id or imdb_id, info_hash)
+
+    try:
+        torrent_id: str = downloader.add_torrent(info_hash)
+        torrent_info: TorrentInfo = downloader.get_torrent_info(torrent_id)
+        session_manager.update_session(session.id, torrent_id=torrent_id, torrent_info=torrent_info, containers=container)
+    except Exception as e:
+        background_tasks.add_task(session_manager.abort_session, session.id)
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)